### DISCUSSION TITLE
Make favorite light colors global

### DISCUSSION THREAD

**Discussion by rednebmas** [home-assistant/frontend#23033] [Other feature requests]
Hello,

I haven't contributed to HomeAssistant before, but I'm interested and have a few questions. I would really like the "favorite colors" palette for lights to be global instead of per light, and it seems [others would like this too.](https://www.reddit.com/r/homeassistant/comments/18gonrr/global_default_favorite_colors/) I am happy to make the change, but I don't want to waste my time making a change to the core source code that will be rejected when I could do it via an addon. 

Would the HA community be open to making the light palette colors global and remove the per light settings? Or a toggle in settings to make it global/individual? Currently it is [saved per entity](https://github.com/home-assistant/frontend/blob/dev/src/dialogs/more-info/components/lights/ha-more-info-light-favorite-colors.ts#L72). What is the process for getting this feature adjustment "approved"?

Additionally, if this is something I should just go ahead and build, I'd be happy to take any suggestions on where to store this information, I'm guessing somewhere in the [front end data](https://developers.home-assistant.io/docs/frontend/data/)?

Cheers,
Sam

**Comment by f3ndot** [comment]
I think this would be very useful. The UI UX could be something like this. Bear in mind the terms "global" and "local" are just placeholders... Not sure how to best articulate the two types.

My preference is Option 3 I think, as its the cleanest with minimal impacts to the current UX.

## Option 1

Keep everything the same but expose an "Edit favorite colors" dropdown. 

<img width="390" src="https://github.com/user-attachments/assets/666697b4-250e-4e72-9f84-dac0354cbafe" />

## Option 2

Keep everything the same but when you enter "Edit colors" mode, and you have the wiggles, tapping the plus/"+" button or existing favorite color will first prompt you with a popup/drawer/whatever with "Do you want a global or local color"?

<img width="390" src="https://github.com/user-attachments/assets/055d8ec1-8a73-4117-9288-49ef850a053d" />


## Option 3

Keep everything the same but when editing a favourite colour you simply have two Save buttons at the bottom: one for just regular "Save" and then a "Save globally". Editing existing local colours would still expose this and tapping "Save globally" in that case would convert or switch them:

<img width="390" src="https://github.com/user-attachments/assets/e15de7f1-5eb0-49b4-9619-8831ea33029f" />

## Visual Differentiation

Additionally to the 3 options we could differentiate between a global and local colour with a rounded square instead of a circle:

<img width="196" alt="Screenshot 2025-05-13 at 9 18 47 AM" src="https://github.com/user-attachments/assets/25a269d6-21fd-4f34-86be-d0cde58704a0" />

**Comment by j9brown** [comment]
I like the ideas to improve the UI for favorite colors.  In the meantime, I implemented this integration to centralize the configuration of favorite colors: https://community.home-assistant.io/t/scenery-better-light-profiles-favorite-colors-and-scenes/883381
